webapi/src/main/scala/org/knora/webapi/slice/search/api/SearchEndpoints.scala

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright © 2021 - 2023 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.knora.webapi.slice.search.api
+
+import org.apache.pekko.http.scaladsl.server.Route
+import sttp.model.HeaderNames
+import sttp.model.MediaType
+import sttp.tapir.*
+import zio.Task
+import zio.ZLayer
+
+import org.knora.webapi.messages.admin.responder.usersmessages.UserADM
+import org.knora.webapi.responders.v2.SearchResponderV2
+import org.knora.webapi.slice.common.api.KnoraResponseRenderer.FormatOptions
+import org.knora.webapi.slice.common.api.KnoraResponseRenderer.RenderedResponse
+import org.knora.webapi.slice.common.api.*
+
+final case class SearchEndpoints(baseEndpoints: BaseEndpoints) {
+
+  private val tags       = List("v2", "search")
+  private val searchBase = "v2" / "searchextended"
+
+  private val gravsearchDescription =
+    "The Gravsearch query. See https://docs.dasch.swiss/latest/DSP-API/03-endpoints/api-v2/query-language/"
+
+  val postGravsearch = baseEndpoints.withUserEndpoint.post
+    .in(searchBase)
+    .in(stringBody.description(gravsearchDescription))
+    .in(ApiV2.Inputs.formatOptions)
+    .out(stringBody)
+    .out(header[MediaType](HeaderNames.ContentType))
+    .tags(tags)
+    .description("Search for resources using a Gravsearch query.")
+
+  val getGravsearch = baseEndpoints.withUserEndpoint.get
+    .in(searchBase / path[String].description(gravsearchDescription))
+    .in(ApiV2.Inputs.formatOptions)
+    .out(stringBody)
+    .out(header[MediaType](HeaderNames.ContentType))
+    .tags(tags)
+    .description("Search for resources using a Gravsearch query.")
+
+  val postGravsearchCount = baseEndpoints.withUserEndpoint.post
+    .in(searchBase / "count")
+    .in(stringBody.description(gravsearchDescription))
+    .in(ApiV2.Inputs.formatOptions)
+    .out(stringBody)
+    .out(header[MediaType](HeaderNames.ContentType))
+    .tags(tags)
+    .description("Count resources using a Gravsearch query.")
+
+  val getGravsearchCount = baseEndpoints.withUserEndpoint.get
+    .in(searchBase / "count" / path[String].description(gravsearchDescription))
+    .in(ApiV2.Inputs.formatOptions)
+    .out(stringBody)
+    .out(header[MediaType](HeaderNames.ContentType))
+    .tags(tags)
+    .description("Count resources using a Gravsearch query.")
+}
```

Comment:
Would it make sense to distinguish somehow POST and GET endpoints descriptions?

Reply:
Why would we, they both do exactly the same and execute the identical code.